python/tank/platform/engine.py

```diff
@@ -467,45 +464,31 @@ def __clear_busy(self):
             self.__global_progress_widget.close()
             self.__global_progress_widget = None
 
-    def log_metric(self, action, log_once=False):
-        """Log an engine metric.
-
-        :param action: Action string to log, e.g. 'Init'
-        :param bool log_once: ``True`` if this metric should be ignored if it
-            has already been logged. Defaults to ``False``.
-
-        Logs a user activity metric as performed within an engine. This is
-        a convenience method that auto-populates the module portion of
-        ``tank.util.log_user_activity_metric()``
-
-        Internal Use Only - We provide no guarantees that this method
-        will be backwards compatible.
-
-        """
-
-        # the action contains the engine and app name, e.g.
-        # module: tk-maya
-        # action: tk-maya - Init
-        full_action = "%s %s" % (self.name, action)
-        util_log_user_activity_metric(self.name, full_action, log_once=log_once)
-
     def log_user_attribute_metric(self, attr_name, attr_value, log_once=False):
-        """Convenience class. Logs a user attribute metric.
-
-        :param attr_name: The name of the attribute to set for the user.
-        :param attr_value: The value of the attribute to set for the user.
-        :param bool log_once: ``True`` if this metric should be ignored if it
-            has already been logged. Defaults to ``False``.
-
-        This is a convenience wrapper around
-        `tank.util.log_user_activity_metric()` that prevents engine subclasses
-        from having to import from `tank.util`.
+        """
+        This method is deprecated and shouldn't be used anymore.
+        """
+        pass
 
-        Internal Use Only - We provide no guarantees that this method
-        will be backwards compatible.
+    @property
+    def _metric_properties(self):
+        """
+        :returns: A dictionary with metric properties specific to this engine.
+        """
+        # Always create a new dictionary so the caller can safely modify it.
+        return {
+            EventMetric.KEY_ENGINE: self.name,
+            EventMetric.KEY_ENGINE_VERSION: self.version,
+            EventMetric.KEY_HOST_APP: self.host_info[0],
+            EventMetric.KEY_HOST_APP_VERSION: self.host_info[1],
+        }
 
+    def _get_metrics_context(self):
+        """
+        :returns: A dictionary with properties to use when emitting a metric
+                  event for this engine.
         """
-        util_log_user_attribute_metric(attr_name, attr_value, log_once=log_once)
+        return self._metric_properties
```

Comment:
is the fact that this method just returns a property an indication that it might not be needed?